src/stan/math/prim/mat/prob/multi_normal_rng.hpp

```diff
@@ -20,10 +20,10 @@ namespace stan {
   namespace prob {
     using Eigen::Dynamic;
 
-    template <class RNG>
+    template <class RNG, typename T1, typename T2>
     inline Eigen::VectorXd
-    multi_normal_rng(const Eigen::Matrix<double, Dynamic, 1>& mu,
-                     const Eigen::Matrix<double, Dynamic, Dynamic>& S,
+    multi_normal_rng(const Eigen::MatrixBase<T1>& mu,
```

Comment:
T1, T2 might be MatrixXd or Map<MatrixXd> but they are certainly not a scalar type.